docs/build.md

```diff
@@ -0,0 +1,129 @@
+- **[flatc](https://github.com/google/flatbuffers)** - used for compiling ztoc
+flatbuffer file and generating corresponding go code.
+
+For fuse/zlib/gcc, they can be installed by your Linux package manager (e.g., `yum` or `apt-get`).
+
+For flatc, you can download and install a [release](https://github.com/google/flatbuffers/releases)
+into your `/usr/local` (or other `$PATH`) directory. For example:
+
+```shell
+wget -c https://github.com/google/flatbuffers/releases/download/v23.3.3/Linux.flatc.binary.g++-10.zip
+sudo unzip Linux.flatc.binary.g++-10.zip -d /usr/local
+rm Linux.flatc.binary.g++-10.zip
+```
+
+## Build soci-snapshotter
+
+First you need `git` to clone the repository (if you intend to contribute, you
+can fork the repository and clone your own fork):
+
+```shell
+git clone https://github.com/awslabs/soci-snapshotter.git
+cd soci-snapshotter
+```
+
+soci-snapshotter uses `make` as the build tool. Assuming you're in the root directory
+of the repository, you can build soci-snapshotter by:
+
+```shell
+make
+```
+
+This builds the project binaries into the `./out` directory. You can install them
+to a `PATH` directory (`/usr/local/bin`) with:
+
+```shell
+sudo make install
+# check soci can be found in PATH
+sudo soci --help
+```
+
+When changing the ztoc's flatbuffer definition, you need to regenerate the generated
+code package with:
+
+> It's rare to make such a change, especially delete a field which is a breaking
+> change and discouraged by flatbuffers.
+
+```shell
+make flatc
+```
+
+## Test soci-snapshotter
+
+We have unit tests and integration tests as part of our automated CI, as well as
+benchmark tests that can be used to test the performance of soci-snapshotter. You
+can run these tests using the following `Makefile` targets:
+
+- `make test`: run all unit tests.
+- `make integration`: run all integration tests.
+- `make benchmarks` (experimental): run all benchmark tests. This requires some
+setup and preparation for public images with SOCI index which are used for benchmarking.
+It is tracked in [#245](https://github.com/awslabs/soci-snapshotter/issues/245).
+
+To speed up develop-test cycle, you can run individual test(s) by utilizing `go test`'s
+`-run` flag. For example, suppose you only want to run a test named `TestFooBar`, you can:
+
+```shell
+# 1. if TestFooBar is a unit test
+GO_TEST_FLAGS="-run TestFooBar" make test
+
+# 2. if TestFooBar is an integration test
+GO_TEST_FLAGS="-run TestFooBar" make integration
+```
+
+## Contribute your change
+
+If you intend to contribute your change, you need to do some checks before
+making a PR (e.g., ensure your code is formatted, your commit is signed, etc).
+For convenience, we provide a script to install these checking tools (`golangci`
+, `ltag`, `git-validation`):
+
+```shell
+./scripts/install-check-tools.sh
+```
+
+Then you can do all the checks by:
+
+```shell
+make check
+```
+
+> We also have a `./scripts/add-ltag.sh` to help add ltag headers to new files.
```

Comment:
I think we should add something before checking with what you're likely going to need to do.

e.g.
```
When you're ready to submit your changes, there are a few steps you should follow to ensure your change is ready.

1. If you added any new files, be sure they contain the SOCI license header
./scripts/add-ltag.sh can do this.

2. Make sure you've run gofmt and that `make test` and `make integration` pass.

3. When you commit your changes, make sure your commit is signed `git commit -s`

As a final check before you submit your PR, verify that `make check` succeeds.
```

I'm leaving out a lot of information, but I think we "Contribute your change" should start with how to make sure your change is ready, not how to check if your change is ready.

Reply:
updated